Assignments/Program.cs

```diff
@@ -1,3 +1,202 @@
+                       greenList = new ();    // List to collect correct and correctly placed letters.
+   public string[] dictWords, guessWords;     // To store loaded strings
+   public StreamWriter testOutputFile = null; // Created for testing purpose.
+
+   public void PlayTheGame () {
+      SetTheBoard ();
+      DisplayTheBoard ();
+      while (!gameOver) {
+         ConsoleKeyInfo key = Console.ReadKey (true);
+         UpdateGameState (key.Key);
+         DisplayTheBoard ();
+      }
+      Console.CursorTop = 28;
+   }
+
+   /// <summary>Initializes the game</summary>
+   public void SetTheBoard () {
+      text = "";
+      (originX, originY) = testing ? (49, 3) : (Console.CursorLeft + (Console.WindowWidth / 2) - 13, Console.CursorTop);
+      dictWords = LoadStrings ("dict-5.txt");
+      guessWords = LoadStrings ("puzzle-5.txt");
+      Console.OutputEncoding = Encoding.Unicode;
+      if (!testing) Console.CursorVisible = false;
+      // Creating Type Board elements:
+      for (int i = 0; i < 6; i++)
+         word.Add (new List<char> { '\u00b7', '\u00b7', '\u00b7', '\u00b7', '\u00b7' });
+      // Generating a random 5 letter word from loaded strings file.
+      var r = new Random ();
+      secretWord = guessWords[r.Next (guessWords.Length)];
+   }
+
+   /// <summary>Displays the updated game board</summary>
+   public void DisplayTheBoard () {
+      if (testing) {
+         // TYPING BOARD:
+         testOutputFile.Write (string.Join ("", Enumerable.Repeat ("\r\n", 3)));
+         if (text.Length == 1) testOutputFile.WriteLine ($"Try-{triesCount + 1}");
+         testOutputFile.Write (new string (' ', originX));
+         for (int i = 0; i < word.Count; i++) {
+            for (int j = 0; j < word[i].Count; j++) {
+               var (a, b, c) = (' ', ' ', word[i][j]);
+               if (i < triesCount) {
+                  // Green --> {} | Blue  --> [] | Grey  --> ()
+                  if (secretWord[j] == c) { (a, b) = ('{', '}'); greenList.Add (c); } else if (secretWord.Contains (c)) { (a, b) = ('[', ']'); blueList.Add (c); } else { (a, b) = ('(', ')'); grayList.Add (c); }
+               }
+               testOutputFile.Write ($" {a}{c}{b} ");
+            }
+            testOutputFile.Write (string.Join ("", Enumerable.Repeat ("\r\n", 3)));
+            testOutputFile.Write (new string (' ', originX));
+         }
+         // KEYBOARD:
+         testOutputFile.WriteLine ();
+         testOutputFile.Write (new string (' ', originX - 3));
+         testOutputFile.WriteLine (new string ('_', 31));
+         testOutputFile.WriteLine ();
+         testOutputFile.Write (new string (' ', originX - 5));
+         for (char c = 'A'; c <= 'Z'; c++) {
+            var (a, b) = greenList.Contains (c) ? ('{', '}')
+                        : blueList.Contains (c) ? ('[', ']')
+                        : grayList.Contains (c) ? ('(', ')')
+                        : (' ', ' ');
+            testOutputFile.Write ($" {a}{c}{b} ");

```

Comment:
Why do we need a list of chars for each color here? Better way is to have a list of tuples where each tuple can carry the char and the color with which it has to be drawn.

Reply:
I tried with Dictionary instead of tuple.